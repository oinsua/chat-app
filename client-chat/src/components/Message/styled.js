import styled from '@emotion/styled';

export const DivPathAutor = styled.div`
                                    display: flex;
                                    justify-content: flex-end;
                                    align-items: center;
                                    font-family: Verdana, Geneva, Tahoma, sans-serif;
                                    font-size: 0.8rem;
                                    width: 100%;                                  
`

export const DivPathFriend = styled.div`
                                    display: flex;
                                    justify-content: flex-start;
                                    align-items: center;
                                    font-family: Verdana, Geneva, Tahoma, sans-serif;
                                    font-size: 0.8rem;
                                    width: 100%;                                  
`

export const DivAutor = styled.div`
                                background-color: var(--brand-color_9);
                                width: 200px;
                                border-radius: 10px;
                                padding: 3px;
                                margin: 5px;
`

export const DivFriend = styled.div`
                                background-color: var(--brand-color_10);
                                width: 200px;
                                border-radius: 10px;
                                padding: 3px;
                                margin: 5px;
`