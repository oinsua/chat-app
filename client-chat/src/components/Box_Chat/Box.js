import React from 'react';
import socket from '../../socket/socket';
//Importando styled 
import {DivForm, DivInput, SpanError, Button, Input, Form}  from './styled';

const Box = ({params}) => {
   
    const handleSubmit = (e) => {
        e.preventDefault();
        const message = {
            user: params.username,
            text: e.target.msg.value
        }
        if(e.target.msg.value.length !== 0){
             socket.emit('send_message', message,(error) => {
                console.log('error desde server: ', error)
              });  
              e.target.msg.value = '';   
       }
}

    const handleInputChange = (e) => {
        e.target.name = e.target.value;
    }

    const handleKeyPress = (e) => {
        
        if(e.key === 'Enter' && e.target.value.length !== 0){
            e.preventDefault();
        const message = {
            user: params.username,
            text: e.target.value
        }
        socket.emit('send_message', message,(error) => {
            console.log('error desde server: ', error)
          });
          e.target.value = ''; 
        }
       

    }

    return (
            <DivForm>
                <Form onSubmit={handleSubmit}>
                    <DivInput>
                        <Input  id="text" name="msg" cols="20" rows="3"  onChange={handleInputChange} onKeyPress={handleKeyPress} required/>
                        <Button name="send" value="Send" type="submit">SEND</Button>
                    </DivInput>
                </Form>
                <SpanError>
                    
               </SpanError>
            </DivForm>
    )
}
export default Box;
