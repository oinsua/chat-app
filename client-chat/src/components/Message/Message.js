import React from 'react';
import {DivPathAutor, DivPathFriend, DivFriend, DivAutor} from './styled';

const Message = ({message: {user, text}, username}) => {
    let isUser = false;
    
    if(user === username)
        isUser = true;
        
    return (
        isUser 
              ? 
               (
                  <DivPathAutor>
                     <span>{user}:</span>
                     <DivAutor>
                      <span>{text}</span>
                     </DivAutor>
                  </DivPathAutor>
               ) 
             : 
               (
                  <DivPathFriend>
                     <DivFriend>
                       <span>{text}</span>
                     </DivFriend>
                     <span>: {user}</span>
                  </DivPathFriend>
               )
    )
}

export default Message;
