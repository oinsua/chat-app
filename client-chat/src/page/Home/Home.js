import React, { useState, useRef, useEffect} from 'react'
import {Helmet} from "react-helmet";
import io from 'socket.io-client';
//Importando hooks
import { useHistory } from "react-router-dom";
//Importando styled
import {DivForm, DivInput, SpanError, Input, Button}  from './styled';
//Importando el favicon
import favicon from '../../asset/img/favicon.png';

const Home = () => {

     /*Iniciando los hook que se van a utilizar */
     const history = useHistory(); //Se crea un objeto para la navegacion
     const inputRef = useRef(null);
     const [data, setData] = useState({username: '', room: ''});
     const [error, setError] = useState(false);
    //Evento que se encarga de enviar el formulario a la Chat
    const handleSubmit =  (e) => {
        e.preventDefault();
        const socket = io('localhost:4000');
        const {username, room} = data;
        socket.emit('join', {username, room} ,(error) => {
           error ? setError(error) : history.push(`/chats/${data.username}/${data.room}`) 
        });        
    }
    //Evento para mantener los input controlados
    const handleInputChange = (e) => {
        console.log(e.target.value)
        setData({
            ...data,
            [e.target.name]: e.target.value
        })
    }
   //Efecto para colar el focus en el primer input
    useEffect(() => {
       inputRef.current.focus();
    }, [])
 
    
    return (
        <>
             <Helmet>
                <title>Home_Web_Chat</title>
                <meta name="application-name" content="home_web_chat" />
                <meta name="description" content="home_web_chat"/>
                <meta name="google" content="notranslate"/>
                <link rel="icon" href={ favicon } sizes="32x32" type="image/png"></link>
             </Helmet>
            <DivForm>
            <form onSubmit={handleSubmit}>
                    <DivInput>
                        <Input  name="username" type="text"  onChange={handleInputChange} ref={inputRef}/>
                        <Input  name="room" type="text"  onChange={handleInputChange}/>
                        <Button name="send" value="Send" type="submit">SEND</Button>
                    </DivInput>
                    <SpanError>
                        <SpanError>{error}</SpanError>
                    </SpanError>
            </form>
            </DivForm>
        </>
    )
}

export default Home
