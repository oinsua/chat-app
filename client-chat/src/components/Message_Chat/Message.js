import React from 'react';
import Message from '../Message/Message';
//Importando Estilos
import {DivMsg, DivMessage} from './styled';

const Messages = ({params, messages}) => {
    //const {username, room} = params;
    console.log('dentro messages : ', messages);
    return (
        <DivMsg> 
            {
                messages.map((message, index) => (<DivMessage key={index}><Message message={message} username={params.username}/></DivMessage>))
            }  
        </DivMsg>
    )
}

export default Messages
