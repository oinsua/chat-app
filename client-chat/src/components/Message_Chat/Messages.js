import React, {useEffect, useRef} from 'react';
import Message from '../Message/Message';
//Importando Estilos
import {DivMsg, DivMessage} from './styled';

const Messages = ({params, messages}) => {

    /*Efecto que se encarga de mantener constantemente subiendo el Div */
    const divRef = useRef(null);
    useEffect(() => {
        divRef.current.scrollIntoView({behavior: 'smooth'})
    })

    return (
        <DivMsg> 
            {
                messages.map((message, index) => (<DivMessage key={index}><Message message={message} username={params.username}/></DivMessage>))
            }
          <div ref={divRef}></div>  
        </DivMsg>
    )
}

export default Messages
