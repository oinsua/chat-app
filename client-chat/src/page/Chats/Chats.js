import React, {useEffect, useState} from 'react';
import socket from '../../socket/socket';
import {Helmet} from "react-helmet";
import { useParams} from "react-router-dom";
//Importando Compoenentes
import Box from '../../components/Box_Chat/Box';
import Messages from '../../components/Message_Chat/Messages';
import Header from '../../components/Header_Chat/Header';
//Importar Estilos
import {DivChat, DivContainer} from './styled';
//Importando el favicon
import favicon from '../../asset/img/favicon.png'

const Chats = () => {
    /*Tomando el parametro enviado por la url */
    const params = useParams();
    const [messages, setMessages] = useState([]);
    /*Efecto para enviar los datos que se necesitan para inicializar el chat */
    useEffect(() => {
      const {username, room} = params; //Se reciben los parametros por la url a traves de router
    
      socket.emit('loading', {username, room}, (error) => { //Emite el evento loading
        console.log('error desde server: ', error)
      });
    
      return () => { 
          socket.on('disconnected'); //Se recibe el evento desconectar
          socket.off(); //Se cierre el enlace
      }
    },[params])
    /*Efecto que se encarga de recibir los mensajes enviados desde el server */
    useEffect(() => {
        socket.on('message', (message) => {
            setMessages([...messages, message]); //Se asignan los new mensajes
        })
    },[messages])

    return (
        <>
        <Helmet>
                <title>Web_Chat</title>
                <meta name="application-name" content="web_chat" />
                <meta name="description" content="web_chat"/>
                <meta name="google" content="notranslate"/>
                <link rel="icon" href={favicon} sizes="32x32" type="image/png"></link>
        </Helmet>
        <DivContainer>
            <DivChat>
                <Header params={params}/>
                <Messages params={params} messages={messages}/>
                <Box params={params}/>
            </DivChat>
        </DivContainer>
        </>
    )
}

export default Chats
