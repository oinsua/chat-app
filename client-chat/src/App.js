import React  from 'react';
import {Route, Switch} from 'react-router-dom';
//Importar componentes
import Home from './page/Home/Home';
import Chats from './page/Chats/Chats';
import Error from './page/Error/Error';
//Importar estiled components
import {Header, Link, Img, H1, DIV} from './styled';
//Importar archivos
import Logo from './asset/img/logo.png';
 
const App = () => {
    return ( 
        <> 
           <DIV>
                <Header> 
                    <Link to="/">
                        <H1>Web</H1>
                        <Img src={Logo} alt="Logo" />
                        <H1>Chat</H1>
                    </Link>
                </Header>
           </DIV>
            <Switch>
                <Route path="/" exact><Home/></Route>
                <Route path="/chats/:username/:room"><Chats/></Route>
                <Route path="/error"><Error/></Route>
            </Switch>
        </>
    );
}

export default App;
