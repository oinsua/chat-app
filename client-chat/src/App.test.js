import { render, screen } from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import App from './App';
import Chats from './page/Chats/Chats';
import Home from './page/Home/Home';
import Messages from './components/Message_Chat/Messages';
import Message from './components/Message/Message';
import Box from './components/Box_Chat/Box';
Enzyme.configure({ adapter: new Adapter() });
 
test('renders text Web', () => {
  render(<BrowserRouter><App /></BrowserRouter>);
  const linkElement = screen.getByText(/Web/i);
  expect(linkElement).toBeInTheDocument();
});

test('render Chats correctly', () => {
  const wrapper = shallow(
    <BrowserRouter><Chats/></BrowserRouter> 
);

expect(toJson(wrapper)).toMatchSnapshot();
});


test('render Home correctly', () => {
  const wrapper = shallow(
    <BrowserRouter><Home/></BrowserRouter> 
);

expect(toJson(wrapper)).toMatchSnapshot();
});


test('render Messages correctly', () => {
  const wrapper = shallow(
    <BrowserRouter><Messages/></BrowserRouter> 
);

expect(toJson(wrapper)).toMatchSnapshot();
});

test('render Box correctly', () => {
  const wrapper = shallow(
    <BrowserRouter><Box/></BrowserRouter> 
);

expect(toJson(wrapper)).toMatchSnapshot();
});

test('render Message correctly', () => {
  const wrapper = shallow(
    <BrowserRouter><Message/></BrowserRouter> 
);

expect(toJson(wrapper)).toMatchSnapshot();
});