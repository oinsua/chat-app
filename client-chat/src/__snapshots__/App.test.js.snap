// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Box correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Box />
</Router>
`;

exports[`render Chats correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Chats />
</Router>
`;

exports[`render Home correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Home />
</Router>
`;

exports[`render Message correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Message />
</Router>
`;

exports[`render Messages correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Messages />
</Router>
`;

exports[`renders text Web1 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Chats />
</Router>
`;
